addons/rating/tests/test_controller.py

```diff
@@ -9,17 +9,36 @@ def setUp(self):
         super(TestControllersRoute, self).setUp()
         self.user = new_test_user(self.env, "test_user_1", email="test_user_1@nowhere.com", tz="UTC")
         self.partner = self.user.partner_id
+        self.partner_1 = self.env['res.partner'].create({
```

Comment:
Actually, changed my mind, let us improve this one even further.

First, instead of extending simply "HttpCase", let's extend "HttpCaseWithUserDemo" for convenience.

Then, inside this setUp, change "self.partner_1" to "self.rated_partner".

After that, inside the "test_controller_rating", we will create 2 ratings instead of one:

```py
partner_model_id = self.env['ir.model'].sudo().search([('model', '=', 'res.partner')], limit=1).id
[rating_test_1, rating_test_2] = self.env['rating.rating'].with_user(self.user).create([{
    'res_model_id': partner_model_id,
    'res_model': 'res.partner',
    'res_id': self.rated_partner.id,
    'partner_id': self.partner.id,
    'rating': 3
} for i in range(2)])
```

After that, you can keep the existing test about "submit_feedback" (on "rating_test_1").

Your next test about internal user will stay but authenticate "self.user_demo" instead (still "rating_test_1").
Your next test should authenticate "self.user" and verify that it works, because it is linked to the partner to which we assign the rating.

Then you can do a test with authenticate "None" on "rating_test_2".
And it should work because non-authenticated.
(We create 2 ratings because it will succeed twice, so you can test that it's been consumed for both).

That way we test everything:
- Can't rate if you're not the assigned partner
- Can rate if you're the assigned partner
- Can rate if public

Does that make sense?

Reply:
Hello @awa-odoo ,
Thanks for the explanation.
I tried to add a new test as you suggested.
Can you please take a look?